### ISSUE TITLE
Drop sklearn dependency. Version too specific. Not used

### ISSUE THREAD

**Issue by a3lem** [axolotl-ai-cloud/axolotl#3276]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [x] I searched previous [Ideas in Discussions](https://github.com/axolotl-ai-cloud/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [x] I searched previous [Issues](https://github.com/axolotl-ai-cloud/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

New user here. I tried installing axolotl both on macOS (aarch64) and on Linux (amd64). In both cases the scikit-learn dependency caused headaches. It's pinned to a very specific version that is only available for older versions of Python. What's more, when I grep for 'skearn' or 'scikit-learn' on the cloned repo, I don't see any actual imports. So either I'm missing something, or this dependency isn't even used?

### ✔️ Solution

Remove scikit-learn dependency (unless I'm mistaken about it not being used, in which case apologies!)

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this feature has not been requested yet.
- [x] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by NanoCode012** [comment]
Closed as solved in #3277
